polygeist/include/mlir/Dialect/Polygeist/Analysis/ReachingDefinitionAnalysis.h

```diff
@@ -43,44 +93,57 @@ class ReachingDefinition : public dataflow::AbstractDenseLattice {
   friend raw_ostream &operator<<(raw_ostream &, const ReachingDefinition &);
 
 public:
-  MLIR_DEFINE_EXPLICIT_INTERNAL_INLINE_TYPE_ID(ReachingDefinition)
-
   using AbstractDenseLattice::AbstractDenseLattice;
-  using ModifiersTy = SetVector<Operation *, SmallVector<Operation *>,
-                                SmallPtrSet<Operation *, 2>>;
+  using DefinitionPtr = std::shared_ptr<Definition>;
+
+  struct CompareDefinition {
+    bool operator()(DefinitionPtr lhs, DefinitionPtr rhs) const {
+      return (lhs && rhs) ? std::less<Definition>{}(*lhs, *rhs)
+                          : std::less<DefinitionPtr>{}(lhs, rhs);
+    }
+  };
+  using ModifiersTy = std::set<DefinitionPtr, CompareDefinition>;
+
+  explicit ReachingDefinition(ProgramPoint p);
+
+  /// Construct an unknown definition (represents the incoming definition at an
+  /// entry point).
+  static ReachingDefinition getUnknownDefinition(ProgramPoint p) {
```

Comment:
is this unknown definition the initial ?

Reply:
`getUnknownDefinition` calls `ReachingDefinition(p)`, which `setModifier` for function memref arguments.
The function name and the comment don't seem to be related to the function body. Am I missing something?

Reply:
> is this unknown definition the initial ?

If the program point is a function definition this static member function creates a `ReachingDefinition` object containing initial definitions for all memref parameters the function has. 
Otherwise it creates a `ReachingDefinition` with empty maps.